### PR TITLE
LRO return type is covariant

### PR DESCRIPTION
**Pull request by lmazuel**
For instance, with discriminator the return type will be the base class, but actual class can be any subtypes. It was already done in some SDKs, make it everywhere.

### CODE REVIEW

File: sdk/core/azure-core/azure/core/polling/_poller.py [hunk 1]
```diff
@@ -33,12 +33,12 @@
 from azure.core.tracing.common import with_current_context
 
 
-PollingReturnType = TypeVar("PollingReturnType")
+PollingReturnType_co = TypeVar("PollingReturnType_co", covariant=True)
```
Comment (johanste) [inline sdk/core/azure-core/azure/core/polling/_poller.py:36]: Do we need the reverse Hungarian notation for the type parameter?

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_polling.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def parse_operation_id(location):
 
 
 @runtime_checkable
-class DocumentModelAdministrationLROPoller(Protocol[PollingReturnType]):
+class DocumentModelAdministrationLROPoller(Protocol[PollingReturnType_co]):
```
Comment (johanste) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_polling.py:40]: Actually, why is this a generic type? We only use it for a single return type, no?

Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_polling.py:40]: As of the last preview, which added classifiers, we have 2 different return types

Comment (lmazuel) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_polling.py:40]: Same question I asked @catalinaperalta :). It's used with two types.

Also, the doc would be better maybe?
> DocumentModelAdministrationLROPoller[DocumentDetails]

Comment (lmazuel) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_polling.py:40]: @catalinaperalta we should probably type it `LROPoller` by the way in the operation, and keep the specific impl a detail (not for this PR)

File: sdk/core/azure-core/azure/core/polling/_poller.py (hunk 1 above)
Comment (lmazuel) [reply sdk/core/azure-core/azure/core/polling/_poller.py:36]: @johanste this is recommendation from typing doc and @kristapratico

**Comment by lmazuel** [comment]
Merging, as LoadTesting failure is unrelated to this PR

**Pull request merged**